lib/src/client/common.dart

```diff
@@ -58,10 +58,51 @@ class ResponseFuture<R> extends DelegatingFuture<R>
     return value;
   }
 
-  ResponseFuture(this._call)
-      : super(_call.response
+  ResponseFuture(ClientCall<dynamic, R> call)
+      : _call = call,
+        super(call.response
             .fold(null, _ensureOnlyOneResponse)
             .then(_ensureOneResponse));
+
+  ResponseFuture._clone(Future<R> future, {ClientCall clientCall})
+      : _call = clientCall,
+        super(future);
+
+  ResponseFuture<S> copy<S>(Future<S> future) =>
+      ResponseFuture._clone(future, clientCall: _getClientCall(future, _call));
+
+  /// clientCall maybe be lost when converting from Future to ResponseFuture
+  static ResponseFuture<T> wrap<T>(Future<T> future, {ClientCall clientCall}) {
+    return ResponseFuture._clone(
+      future,
+      clientCall: (_getClientCall(future, clientCall)),
+    );
+  }
+
+  static ClientCall _getClientCall(Future future, [ClientCall fallback]) {
```

Comment:
I think it would be easier to read if you do 

```dart
static ClientCall<dynamic, R> _unwrap<R>(Future<R> future) =>
   future is ResponseFuture ? future._call : null;
```

and then make call-sites `_unwrap(future) ?? clientCall`.

Reply:
SGTM, let me update